gpytorch/variational/nearest_neighbor_variational_strategy.py

```diff
@@ -134,13 +138,25 @@ def __call__(self, x: Tensor, prior: bool = False, **kwargs: Any) -> Multivariat
         if x is not None:
             assert self.inducing_points.shape[:-2] == x.shape[:-2], (
                 f"x batch shape must matches inducing points batch shape, "
-                f"but got train data batch shape = {x.shape[:-2]}, "
+                f"but got x batch shape = {x.shape[:-2]}, "
                 f"inducing points batch shape = {self.inducing_points.shape[:-2]}."
             )
 
         # Delete previously cached items from the training distribution
         if self.training:
             self._clear_cache()
+
+            # (Maybe) initialize variational distribution
+            if not self.variational_params_initialized.item():
+                prior_dist = self.prior_distribution
+                self._variational_distribution.variational_mean.data.copy_(prior_dist.mean)
+                self._variational_distribution.variational_mean.data.add_(
+                    torch.randn_like(prior_dist.mean), alpha=self._variational_distribution.mean_init_std
+                )
+                # initialize with a small variational stddev for quicker conv. of kl divergence
+                self._variational_distribution._variational_stddev.data.copy_(torch.tensor(1e-2))
+                self.variational_params_initialized.fill_(1)
```

Comment:
Ahhh okay, I see you've added a new initialization scheme.

Reply:
Yes. In practice I found that the variational standard deviation tends to shrink towards 0 in the end given that inducing points = data points. If initialized with ones, the KL term is way larger than log likelihood term, taking a longer time to converge. Initializing with a smaller value speeds up the convergence.

Reply:
For the future, it's good to list all of these little changes in the PR description as well :)